alea/runner.py

```diff
@@ -0,0 +1,400 @@
+from copy import deepcopy
+from typing import Callable, Optional
+from datetime import datetime
+from pydoc import locate
+import inspect
+
+from tqdm import tqdm
+import numpy as np
+
+from inference_interface import toydata_from_file, numpy_to_toyfile
+from alea.statistical_model import StatisticalModel
+
+
+class Runner:
+    """
+    Runner manipulates statistical model and toydata.
+        - initialize statistical model
+        - generate or read toy data
+        - save toy data if needed
+        - fit parameters
+        - write output file
+    """
+
+    def __init__(
+            self,
+            statistical_model: str,
+            statistical_model_args: dict = None,
+            parameter_definition: Optional[dict or list] = None,
+            confidence_level: float = 0.9,
+            confidence_interval_kind: str = 'central',
+            confidence_interval_threshold: Callable[[float], float] = None,
+            poi: str = None,
+            hypotheses: list = None,
+            common_generate_values: dict = None,
```

Comment:
What is the distinction between common_generate_values and true_generate_values?
the generate values are always true for the datasets generated with them, no?

Reply:
updated in previous commits.